harvard-lts/epadd-curator-app
# Implement Curator App

**Implement Curator App**
* * *

**Ticket Link**: [epadd#58](https://app.zenhub.com/workspaces/epadd-harvard-62c485cd49f954001312ebdd/issues/harvard-lts/epadd/58) [epadd#100](https://app.zenhub.com/workspaces/epadd-harvard-62c485cd49f954001312ebdd/issues/harvard-lts/epadd/100)

# What does this Pull Request do?
In order to notify DIMS of an ePADD export ready for ingest, the Curator App monitors the ePADD nextcloud buckets for a manifest file. Once found, the curator app collects metadata and calls DIMS to begin the ingest process. The metadata collection will need to be expanded once sidecar files are included in the exports. Once ingest is called, an ingest.txt or ingest.txt.failed is placed in the export directory to indicate ingest status of that export.

# How should this be tested?

1. Deploy to dev
2. Call `http://ltsds-cloud-dev-1.lib.harvard.edu:10586/testExport`
3. Confirm that the following files are uploaded to the dev ePADD nextcloud bucket at the prefix `test/`:
`TESTTRIGGER`
`manifest.txt`
`test_export.zip`
5. After 5 mins, confirm that an `ingest.txt` file exists at `test/ingest.txt`

**Note:** Sometimes the Nextcloud mount doesn't update right away, so when checking for the files it may be quicker to use the aws cli tool

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? No
- integration tests? Yes

# Interested parties
@ives1227 @awoods
